### ISSUE TITLE
Support SurrealDB

### ISSUE THREAD

**Issue by yovanoc** [prisma/prisma#15346]
Support for https://surrealdb.com could be a good idea with the gain in popularity these days.

https://www.youtube.com/watch?v=C7WFwgDRStM

**Comment by LoKy-dev** [comment]
Can't wait to see prisma implementation as the software matures.

**Comment by rvdende** [comment]
I'm looking to perhaps implement this, can anyone point me to where I need to start looking? prisma is a bit of a large codebase.

**Comment by mysticaltech** [comment]
@rvdende On Surreal's side, this is what you need IMHO. https://github.com/surrealdb/surrealdb.js and on this side, IMHO have a look at how they did it for Planetscale!

**Comment by escwxyz** [comment]
Any updates? @rvdende 
 
PS: maybe not relevant, but could check out [cirq](https://github.com/StarlaneStudios/cirql) for now?

**Comment by fjodorsch** [comment]
Hey, what is the plan, now that SurrealDB is in 1.0 and has a node SDK?
https://surrealdb.com/docs/integration/sdks/nodejs

**Comment by janpio** [comment]
No plan right now. SurrealDB is too different to our other supported databases (which all talk a slight variant of standard SQL, or MongoDB), so implementing support for SurrealDB currently would be a _major_ lift. We can not commit the resources to do that at this time.

**Comment by lveillard** [comment]
We are planning to add surrealDB to our ORM (https://github.com/Blitzapps/blitz-orm)

Currently it works with typeDB but surrealDB is the next adapter to be added. Happy to get your requirements to see if they will fit our ORM :)

**Comment by mysticaltech** [comment]
@lveillard You are using AGPL, that's a no go for any project that wants to do anything commercial, too bad.

**Comment by lveillard** [comment]
> @lveillard You are using AGPL, that's a no go for any project that wants to do anything commercial, too bad.

Yes we need to review the licence. Anyway any comercial project using it will have our blessing as long as they help 😂. And we might change the licence later

**Comment by MannyJMusic** [comment]
Would like to throw my support in as well for SurrealDB support.  Very flexible but have been a long time prisma user and think it would be a god send.

**Comment by lveillard** [comment]
Btw shameless plug, our open source ORM already works with surrealDB and TypeDB
https://github.com/Blitzapps/blitz-orm

Obviously not as stable and fast as prisma yet, but is more oriented for graph DBs logic and has some advantages like shared fields, computed hooks, graphql-like batch mutations... 

Also as a little project we are happy to accept any feature you might require as long as we get some help 🙌

**Comment by nicolaeser** [comment]
Just placing a message here for reminding prisma to not fuck with SurrealDB. Never used SurreallDB and just heard the feedback of so many users and that's enough motivation for me to try it. Please support it, would make many lifes so much easier.

**Comment by Bitcreep** [comment]
> Any updates? [@rvdende](https://github.com/rvdende)
> 
> PS: maybe not relevant, but could check out [cirql](https://github.com/StarlaneStudios/cirql) for now?

This project is no-longer being maintained and should not be used in production.

**Comment by nicolaeser** [comment]
Reminder
